### PR TITLE
Add edge case tests for transition_regime_coagulation

### LINKED ISSUE #501: **suggestion (testing):** Consider adding tests for edge cases
**suggestion (testing):** Consider adding tests for edge cases

With the removal of edge case values (0.1 and 0.0) from the test arrays, it might be beneficial to add separate unit tests that specifically target these edge cases to ensure the numerical issue is properly addressed.

```
    np.testing.assert_almost_equal(result_single, expected_single, decimal=4)

    # Test edge cases
    edge_case_input = np.array([0.1, 0.0])
    edge_case_result = transition_regime_coagulation(edge_case_input)
    edge_case_expected = np.array([0.0, 0.0])  # Adjust expected values as needed
    np.testing.assert_almost_equal(edge_case_result, edge_case_expected, decimal=4)
```

_Originally posted by @sourcery-ai[bot] in https://github.com/uncscode/particula/pull/500#discussion_r1799725572_

### PR DESCRIPTION
**Pull request by mahf708**
Fixes #501

Add unit tests for edge cases in `transition_regime_coagulation` function.

* Add a new test function `test_transition_regime_coagulation_edge_cases` in `particula/dynamics/coagulation/tests/transition_regime_test.py`.
* Include edge case values (0.1 and 0.0) in the test function.
* Use `np.testing.assert_almost_equal` to compare results with expected values.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/uncscode/particula/pull/534?shareId=5335750d-dab1-41cb-a5eb-ccf572272e52).

## Summary by Sourcery

Tests:
- Add unit tests for edge cases in the transition_regime_coagulation function, including zero, small, negative, very large, NaN, and infinity values.

### CODE REVIEW

File: particula/dynamics/coagulation/transition_regime.py [hunk 1]
```diff
@@ -30,6 +34,13 @@ def hard_sphere(
     Journal of Chemical Physics, 126(12).
     https://doi.org/10.1063/1.2713719
     """
+    if np.any(diffusive_knudsen < 0):
```
Comment (Gorkowski) [inline particula/dynamics/coagulation/transition_regime.py:37]: probably not the best way to scale input validations, we should think about that going forward

**Review by Gorkowski — APPROVED** [review]
